xds/src/main/java/io/grpc/xds/XdsNameResolverProvider.java

```diff
@@ -41,19 +41,22 @@
 @Internal
 public final class XdsNameResolverProvider extends NameResolverProvider {
 
-  private String scheme = "xds";
-  private Map<String, ?> bootstrapOverride;
+  private static final String SCHEME = "xds";
+  private final String scheme;
+  private final Map<String, ?> bootstrapOverride;
+
+  public XdsNameResolverProvider() {
+    this(SCHEME, null);
+  }
 
   /**
-   * A convenient method to create a {@link XdsNameResolverProvider} with custom scheme and
-   * bootstrap.
+   * A convenient constructor to allow creating a {@link XdsNameResolverProvider} with custom scheme
+   * and bootstrap.
    */
-  public static XdsNameResolverProvider createForTest(String scheme,
```

Comment:
I had intended for the constructor to be private, and for us to still have the "for test" factory method to discourage it for other uses.